lib/std/os/uefi/tables/runtime_services.zig

```diff
@@ -23,50 +25,549 @@ pub const RuntimeServices = extern struct {
+        metadata_size: u32,
+        flags: Flags,
+
+        pub fn payloadConst(self: *const VariableAuthentication3) []const u8 {
+            return @constCast(self).payload();
+        }
+
+        pub fn payload(self: *VariableAuthentication3) []u8 {
+            var ptr: [*]u8 = @ptrCast(self);
+            return ptr[@sizeOf(VariableAuthentication3)..self.metadata_size];
+        }
+
+        pub const Flags = packed struct(u32) {
+            update_cert: bool = false,
+            _pad: u31 = 0,
+        };
+
+        pub const Type = enum(u8) {
+            timestamp = 1,
+            nonce = 2,
+            _,
+        };
+    };
+
+    pub const VariableInfo = struct {
+        max_variable_storage_size: u64,
+        remaining_variable_storage_size: u64,
+        max_variable_size: u64,
+    };
+
+    pub const VariableNameIterator = struct {
+        pub const NextSizeError = uefi.UnexpectedError || error{
+            DeviceError,
+            Unsupported,
+        };
+
+        pub const IterateVariableNameError = NextSizeError || error{
+            BufferTooSmall,
+            NotFound,
+        };
+
+        services: *const RuntimeServices,
+        buffer: []u16,
+        guid: Guid,
+
+        pub fn nextSize(self: *VariableNameIterator) NextSizeError!?usize {
+            var len: usize = 0;
+            switch (self.services._getNextVariableName(
+                &len,
+                null,
+                &self.guid,
+            )) {
+                .success, .buffer_too_small => return len,
+                .not_found => return null,
+                .device_error => return Error.DeviceError,
+                .unsupported => return Error.Unsupported,
+                else => |status| return uefi.unexpectedStatus(status),
+            }
+        }
+
+        /// Call `nextSize` to ensure that `buffer` is large enough to hold the next
+        /// variable name.
+        pub fn next(
+            self: *VariableNameIterator,
+        ) IterateVariableNameError![:0]const u16 {
+            var len = self.buffer.len;
+            switch (self.services._getNextVariableName(
+                &len,
+                @ptrCast(self.buffer.ptr),
+                &self.guid,
+            )) {
+                .success => return self.buffer[0 .. len - 1 :0],
+                .not_found => return Error.NotFound,
+                .buffer_too_small => return Error.BufferTooSmall,
+                .device_error => return Error.DeviceError,
+                .unsupported => return Error.Unsupported,
+                else => |status| return uefi.unexpectedStatus(status),
+            }
+        }
+    };
+
+    pub const MonotonicCounter = struct {
+        services: *const RuntimeServices,
+        count: u32 = 0,
+
+        pub fn next(self: *MonotonicCounter) GetNextHighMonotonicCountError!u32 {
+            try self.services.getNextHighMonotonicCount(&self.count);
+            return self.count;
+        }
+    };
```

Comment:
Why does this exist, it will always return the same value during any single execution, the high monotonic count only increases on system reset.